modules/export_model.py

```diff
@@ -881,17 +884,23 @@ def make_mesh_object(operator, context, mesh: bpy.types.Object, group_name, i, m
         # ssbh_data expects all colors to be 32 bit floats in the range 0.0 to 1.0.
         # TODO: Support other data types.
         if attribute.data_type == 'FLOAT_COLOR':
-            loop_colors = np.zeros(len(mesh.data.loops) * 4, dtype=np.float32)
+            if attribute.domain == 'POINT':
+                loop_colors = np.zeros(len(mesh.data.vertices) * 4, dtype=np.float32)
```

Comment:
This can be split into two blocks to avoid repetition. Unrecognized domains should raise a runtime error similar to unrecognized data types.
```python
if attribute.domain == 'CORNER':
    colors = np.zeros(...)
elif attribute.domain == 'POINT':
    colors = np.zeros(...)
else:
    raise RuntimeError(...)

if attribute.data_type == 'FLOAT_COLOR':
    attribute.data.foreach_get('color', colors)
elif attribute.data_type == 'BYTE_COLOR':
     # Despite being called 'BYTE' color this uses an array of 4 floats.
    attribute.data.foreach_get('color', colors)
else:
    raise RuntimeError(...)

if attribute.domain == 'CORNER':
    colors = per_loop_to_per_vertex(...)
ssbh_color_layer.data = colors

```